addons/stock/models/stock_lot.py

```diff
@@ -16,6 +16,9 @@ class StockLot(models.Model):
     _check_company_auto = True
     _order = 'name, id'
 
+    def _read_group_location_id(self, locations, domain, order):
+        return locations.search(['|', ('usage', 'in', ('customer', 'supplier')), '&', ('usage', '=', 'internal'), ('location_id.usage', '=', 'view')], order=order)
+
```

Comment:
I'm not sure the condition ('usage', '=', 'internal'), ('location_id.usage', '=', 'view') is a good idea.
If the user has a good configuration with many rows in the warehouse of type view and drawer as real internal location, you will see too many data. Instead I would use a search on warehouse and take their lot_stock_id